sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -267,32 +256,26 @@ def _sample_posterior_mcmc(
         thin: int = 10,
         warmup: int = 20,
         num_chains: Optional[int] = 1,
-        show_progressbar: Optional[int] = True,
+        show_progressbar: bool = True,
     ) -> Tensor:
         r"""
         Return MCMC samples from posterior $p(\theta|x)$.
 
         Args:
-            x: conditioning context for posterior $p(\theta|x)$
-            
-            num_samples: desired output samples
-            
-            mcmc_method: one of 'metropolis-hastings', 'slice', 'hmc', 'nuts'
-            
-            thin: thinning factor for the chain, e.g. for thin=3 only every
-                third sample will be returned, until a total of num_samples
-
-            show_progressbar: whether to show a progressbar during sampling
+            x: Conditioning context for posterior $p(\theta|x)$.
+            num_samples: Desired number of samples.
+            mcmc_method: One of `metropolis-hastings`, `slice`, `hmc`, `nuts`.
```

Comment:
* @jan-matthis needs more time to make sure that pyro is up to the task in every situation, so we were leaving this for post-release.
* my impression is that [line 369](https://github.com/mackelab/sbi/blob/06361d0d6754d7ba77345aed00391bc2bb58798d/sbi/inference/posteriors/sbi_posterior.py#L369) indicates that we are not supporting Metropolis Hastings in Pyro as it is currently. So I will remove it from the docstring.

Reply:
- so let's add `slice-np` as an option here, right?
- great!

Reply:
You know there was a bug here... `slice_np` was never used because we had `if mcmc_method == "slice-np"` (dash) and the other branch was a catch-all. Going to run tests now, I cross fingers that this was not shadowing bugs in `slice_np`.,,

Reply:
Good catch, let's fix this! But luckily the defaults in `snpe_c`, `snpe_base`, and `sbi_posterior.init()` are all correct, so the wrong default in `_sample_posterior_mcmc()` should be overwritten. But yes the warning would not have been working.